src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/reporting/generator/BedUtilisationReportGenerator.kt

```diff
@@ -103,8 +107,10 @@ class BedUtilisationReportGenerator(
         effectiveTurnaroundDays = effectiveTurnaroundDays,
         voidDays = voidDays,
         totalBookedDays = totalBookedDays,
-        totalDaysInTheMonth = daysInMonth,
-        occupancyRate = totalBookedDays.toDouble() / daysInMonth,
+        bedspaceStartDate = this.createdAt.toLocalDate(),
+        bedspaceEndDate = this.endDate,
+        bedspaceOnlineDays,
+        occupancyRate = totalBookedDays.toDouble() / bedspaceOnlineDays,
```

Comment:
the ticket mentions the following report filter changes. I couldn't square them with the existing query but i may be misunderstanding the requirement. Does this need doing?

```The report needs to only include Bedspaces where the following is true:

Bedspace Start Date (i.e. Creation Date) BEFORE Report End Date

AND (Bedspace End Date is AFTER Report Start Date OR Bedspace End Date is <blank>)

```

Reply:
This is the criteria to calculate bedspaceOnlineDays calculate the  which is in line 88